Tests/Common/LoggerMockExtensions.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.Tests.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using Microsoft.Extensions.Logging;
+    using Moq;
+
+    public sealed record LoggerLogInvocation(LogLevel LogLevel, EventId EventId, Exception? Exception, string Message);
+
+    public static class LoggerMockExtensions
+    {
+        private static readonly MethodInfo? LogMethodInfo = typeof(ILogger).GetMethod(nameof(ILogger.Log));
+
+        public static IEnumerable<LoggerLogInvocation> GetLogInvocations(this Mock<ILogger> mock)
+        {
+            ArgumentNullException.ThrowIfNull(mock, nameof(mock));
```

Comment:
Why pass the second argument to `ArgumentNullException.ThrowIfNull`? Even when omitting it it should automatically pick up the correct parameter name, no?

Reply:
Right, completely forgot it has the new `CallerArgumentExpression` applied. 🤦🏽‍♂️ Removed with 4949ae882941f222a303acf37cccc5ffac52905c.